### PR TITLE
Validate upgrade resource and client service

### LINKED ISSUE #7222: [WebSocket] Need to validate HTTP to WebSocket Upgrade Resource in Compile Time
**Description:**
Right now Http to WebSocket upgrade resource is not validated. This should be validated in the compile time in order to work the functionality correct.

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java [hunk 1]
```diff
@@ -31,24 +33,25 @@
  */
 public class WebSocketResourceValidator {
 
-    public static void validate(String serviceName, BLangResource resource, DiagnosticLog dlog) {
+    public static void validate(String serviceName, BLangResource resource, DiagnosticLog dlog, boolean client) {
```
[the remark this answers is not in the document]
Comment (riyafa) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java]: It is not the best practice to use is in variable names. It is used in the method names.

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java [hunk 2]
```diff
@@ -58,72 +61,85 @@ public static void validate(String serviceName, BLangResource resource, Diagnost
...
             dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos,
-                               "onOpen resource is not supported for the " + serviceName);
+                               "onOpen resource is not supported for " + WebSocketConstants.WEBSOCKET_CLIENT_SERVICE +
+                                       " " + serviceName);
         }
     }
 
-    private static void validateOnTextResource(String serviceName, BLangResource resource, DiagnosticLog dlog) {
+    private static void validateOnTextResource(String serviceName, BLangResource resource, DiagnosticLog dlog,
+                                               boolean client) {
```
[the remark this answers is not in the document]
Comment (riyafa) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java]: same as above

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (irunika) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java]: But "client" is not a verb. To use like that should be a verb. Unless it implies an object IMO.
WDYT?

[the remark this answers is not in the document]
Comment (riyafa) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java]: It need not be a verb[1]. Where did you get that idea from? 
[1] https://stackoverflow.com/q/5322648/3599535

**Pull request merged**